Basefee reduction (#125)
avalanche-foundation/ACPs: ACPs/125-basefee-reduction/README.md

```text
ACP: 125

> Title: Reduce C-Chain base fee to 1 Gwei.

Author: done

Author(s): Stephen Buttolph <https://github.com/StephenButtolph>, Darioush Jalali <https://github.com/darioush>
Status: Proposed
Track: Standards
```

## Abstract

> This ACP proposes reducing the base fee on the Avalanche C-Chain from 25 Gwei to 1 Gwei.

Reviewer: "this ACP" could be dropped throughout this file

## Motivation

With dynamic fees, the gas price is supposed to be a result of a continuous auction such that the consumed gas per second converges to the target gas usage per second.
When dynamic fees were first introduced safeguards were introduced to ensure that the mechanism worked as intended, such as a relatively high minimum gas price and a maximum gas price.

> The maximum gas price has since been entirely removed. The minimum gas price has been reduced significantly. However, it is still too high, and is therefore reducing usage of the network.

Reviewer: There should be an empty line between these

Author: done

Reviewer 2: Instead of asserting that "it is still too high", I'd recommend we say that the fact that the observed base fee is often pinned to this minimum shows that it is higher than what the market demands, and therefore artificially reducing network usage.

## Specification

The dynamic fee calculation currently must enforce a minimum base fee of 25 Gwei.
This ACP proposes reducing the minimum base fee to 1 Gwei upon the next network upgrade activation.

## Backwards Compatibility

This ACP modifies the consensus rules for the C-Chain, therefore it requires a network upgrade.

## Reference Implementation

A draft implementation of this ACP for the coreth VM can be found [here](https://github.com/ava-labs/coreth/pull/604/files).

## Security Considerations

> N/A

Reviewer 3: Not sure how pedantic we want to be here.

This will increase state bloat (as it can make gas cheaper). But not past the parameterized bounds.

Reviewer: I'd include that the dynamic fee algorithm responded appropriately during the inscription era which gives reasonable confidence that this min fee crutch is no longer necessary

Author: added, not sure if others prefer a different way to refer to the "inscription era"

```text
ACP: 125

> Title: Reduce C-Chain min base fee from 25 nAVAX to 1 nAVAX

Reviewer 2: ```suggestion
Title: Reduce C-Chain minimum base fee from 25 nAVAX to 1 nAVAX
```

Author(s): Stephen Buttolph <https://github.com/StephenButtolph>, Darioush Jalali <https://github.com/darioush>
Status: Proposed
Track: Standards
```

## Abstract

> Reduce the min base fee on the Avalanche C-Chain from 25 nAVAX to 1 nAVAX.

Reviewer 2: ```suggestion
Reduce the minimum base fee on the Avalanche C-Chain from 25 nAVAX to 1 nAVAX.
```

## Motivation

With dynamic fees, the gas price is supposed to be a result of a continuous auction such that the consumed gas per second converges to the target gas usage per second.

> When dynamic fees were first introduced safeguards were added to ensure the mechanism worked as intended, such as a relatively high minimum gas price and a maximum gas price.

Reviewer 2: ```suggestion
When dynamic fees were first introduced, safeguards were added to ensure the mechanism worked as intended, such as a relatively high minimum gas price and a maximum gas price.
```

The maximum gas price has since been entirely removed. The minimum gas price has been reduced significantly. However, it is still too high, and is therefore reducing usage of the network.

## Specification

> The dynamic fee calculation currently must enforce a minimum base fee of 25 nAVAX.
> This change proposes reducing the minimum base fee to 1 nAVAX upon the next network upgrade activation.

Reviewer 2: ```suggestion
The dynamic fee calculation currently must enforce a minimum base fee of 25 nAVAX.

This change proposes reducing the minimum base fee to 1 nAVAX upon the next network upgrade activation.
```

## Backwards Compatibility

Modifies the consensus rules for the C-Chain, therefore it requires a network upgrade.

## Reference Implementation

A draft implementation of this ACP for the coreth VM can be found [here](https://github.com/ava-labs/coreth/pull/604/files).

## Security Considerations

> This makes gas cheaper, therefore may increase state bloat.
> 
> We note the dynamic fee algorithm responded appropriately during the periods of high use (such as Dec. 2023), which gives reasonable confidence that enforcing a 25nAVAX minimum fee is no longer necessary.

Reviewer 2: ```suggestion
Lower gas costs may increase state bloat. However, we note that the dynamic fee algorithm responded appropriately during periods of high use (such as Dec. 2023), which gives reasonable confidence that enforcing a 25 nAVAX minimum fee is no longer necessary.
```
